src/pages/Account/DeleteCardForm.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import { compose, graphql } from 'react-apollo';
+
+import { DELETE_PAYMENT_SOURCE, GET_PAYMENT_SOURCES, PaymentSource } from 'networking/paymentSources';
+import { Button, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';
+
+interface Props {
+  paymentSource: PaymentSource;
+  deletePaymentSource: (paymentSourceId: string) => Promise<any>;
+  handleModal: () => void;
+  setAlert: (msg?: string) => void;
+}
+
+const DeleteCardForm = ({ deletePaymentSource, handleModal, paymentSource, setAlert }: Props) => {
+  const handleDelete = () => {
+    return deletePaymentSource(paymentSource.id)
+      .then(() => {
+        handleModal();
+        setAlert(`Success! Your card ${paymentSource.stripeBrand} ending in ${paymentSource.stripeLast4} has been deleted.`);
+      })
+      .catch((error: any) => alert('Oops! Something went wrong:\n\n' + error));
+  }
+  return (
+    <>
+      <ModalHeader>Delete Card</ModalHeader>
```

Comment:
It's a minor point, but for `NewCardForm` the `ModalHeader` and `ModalBody` are added by `AccountPayment`, whereas here they're part of the form component. For consistency, I'd use the approach from `NewCardForm` in both places. (It's also a bit odd that `ModalHeader` and so on are in here without a `Modal`; it's a bit non-intuitive that `DeleteCardForm` depends on providing a wrapper `Modal` externally.)

Reply:
Agreed - consistency should be maintained.

One thing that made me feel uneasy was the `<ModalFooter />` component that goes after the ModalBody has some buttons with networking logic in it. (e.g. handleDelete). The `<NewCardForm />` doesn't even have a `<ModalFooter />` which might go against the framework's intentions.

I'm thinking to just make the <Modal><Form /></Modal> into one component. If the Form needs to be extracted somewhere down the line, it shouldn't be too difficult to pull out.

Reply:
Insightful finding:

After doing some research on Modal implementation on ReactStrap, it looks like the convention is that Modals are supposed to be rendered at all times but use the `isOpen` prop to control them. (It may not matter too much for modals, but for `Alert`, I had trouble with the fade out animation because I was conditionally rendering the whole component).